phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -121,8 +170,21 @@ public synchronized void initScanner() throws SQLException {
                 this.scanIterator =
                         new ScanningResultIterator(htable.getScanner(scan), scanMetrics);
             } catch (IOException e) {
-                Closeables.closeQuietly(htable);
-                throw ServerUtil.parseServerException(e);
+                if(handleSplitRegionBoundaryFailureDuringInitialization) {
```

Comment:
This is required for ChunkedResultIterator if we are going to deprecate or not use it we don't need these changes. Other than that we might need to deal with it in PhoenixRecordReader(currently we skipping range check but for local indexes it's compulsory to handle otherwise we might miss one of the daughter records)